src/lib/encoding/confline.c

```diff
@@ -33,14 +33,23 @@ config_line_append(config_line_t **lst,
                    const char *key,
                    const char *val)
 {
-  tor_assert(lst);
-
   config_line_t *newline;
 
   newline = tor_malloc_zero(sizeof(config_line_t));
   newline->key = tor_strdup(key);
   newline->value = tor_strdup(val);
   newline->next = NULL;
+
+  config_line_append_line(lst, newline);
+}
+
+/** Helper: append <b>newline</b> to the end of <b>lst</b>. */
+void
+config_line_append_line(config_line_t **lst,
```

Comment:
I think this name isn't great. Is the main difference between `config_line_append()` and `config_line_append_line()` that the first one duplicates stuff? Maybe rename the first to `config_line_append_alloc()` and the helper to `config_line_append()`, or `config_line_append()` and `config_line_append_noalloc()`?

Also in terms of pre-existing technical debt, this is a linked list implementation that isn't using the `tor_queue.h` macros. Maybe we should fix that at some point? (I think it's fine if you open a new ticket for this.)

Reply:
Wrt  config_line_append_line, here's an easier solution: I think we can revert the commit that introduced this.  I wrongly thought that we would need this function, but looking through the branch, it seems that we never actually started using it.

I agree that a new ticket is the right place for changing config_line_t to use SLIST.  I've done so with trac#31074.